### ISSUE TITLE
Apache does not respect TLS 1.3 cipher suite order

### ISSUE THREAD

**Issue by janipewter** [oerdnj/deb.sury.org#1112]
**Describe the bug**
I apologise if this is not the correct place to post this. I am using ondrej ppa on Ubuntu 16.04. I have Apache 2.4.38 and OpenSSL 1.1.1b.

I have noticed that Apache does not respect the configured cipher suite order for TLS 1.3. Cipher suite order is respected correctly for TLS 1.2 (all other protocols are disabled). I checked this on another similar server and it does not experience the problem. The only difference with the other server is that it has OpenSSL 1.1.1a (most other configuration and environment is the same)

**To Reproduce**
1. Configure Apache with SSLCipherSuite TLSv1.3 to anything other than TLS_AES_256_GCM_SHA384:TLS_CHACHA20_POLY1305_SHA256:TLS_AES_128_GCM_SHA256
2. Test server with testssl.sh or something like ssllabs.com test
3. Expect cipher order to be as you set it
4. Server always sends order as AES-256, CHACHA20, AES-128 regardless of configuration

**Expected behavior**
Server should respect the cipher suite order when "SSLHonorCipherOrder     on" is set.

**Distribution (please complete the following information):**
 - OS: Ubuntu 16.04.6
 - Architecture: amd64
 - Repository: ondrej ppa

**Package(s) (please complete the following information):**
apache2:
  Installed: 2.4.38-2+ubuntu16.04.1+deb.sury.org+8
  Candidate: 2.4.38-2+ubuntu16.04.1+deb.sury.org+8
  Version table:
 *** 2.4.38-2+ubuntu16.04.1+deb.sury.org+8 500
        500 http://ppa.launchpad.net/ondrej/apache2/ubuntu xenial/main amd64 Packages
        100 /var/lib/dpkg/status
     2.4.23-1ppa1~xenial 500
        500 http://ppa.launchpad.net/linux1488/openssl-chacha20poly1305/ubuntu xenial/main amd64 Packages
     2.4.18-2ubuntu3.9 500
        500 http://gb.archive.ubuntu.com/ubuntu xenial-updates/main amd64 Packages
     2.4.18-2ubuntu3.8 500
        500 http://security.ubuntu.com/ubuntu xenial-security/main amd64 Packages
     2.4.18-2ubuntu3 500
        500 http://gb.archive.ubuntu.com/ubuntu xenial/main amd64 Packages

openssl:
  Installed: 1.1.1-1+ubuntu16.04.1+deb.sury.org+1
  Candidate: 1.1.1-1+ubuntu16.04.1+deb.sury.org+1
  Version table:
 *** 1.1.1-1+ubuntu16.04.1+deb.sury.org+1 500
        500 http://ppa.launchpad.net/ondrej/apache2/ubuntu xenial/main amd64 Packages
        100 /var/lib/dpkg/status
     1.1.0h-2.0+ubuntu16.04.1+deb.sury.org+1 500
        500 http://ppa.launchpad.net/ondrej/php/ubuntu xenial/main amd64 Packages
     1.0.2k-2ppa1~xenial 500
        500 http://ppa.launchpad.net/linux1488/openssl-chacha20poly1305/ubuntu xenial/main amd64 Packages
     1.0.2g-1ubuntu4.15 500
        500 http://gb.archive.ubuntu.com/ubuntu xenial-updates/main amd64 Packages
        500 http://security.ubuntu.com/ubuntu xenial-security/main amd64 Packages
     1.0.2g-1ubuntu4 500
        500 http://gb.archive.ubuntu.com/ubuntu xenial/main amd64 Packages

**Additional context**
The server is serving ECC and RSA keys simultaneously. I can provide a subdomain to show the problem if necessary

**Comment by oerdnj** [comment]
Hi, thanks for the bugreport, I believe that this would be better reported to apache2 bugzilla, as the packages don't do any TLS patching.  I would be welcome if you could report the upstream bugreport if you create it.

Anyway, have you tried googling the subject of this bugreport, it seems that this might do the trick:
https://www.apachelounge.com/viewtopic.php?p=37474

Or this: https://ayesh.me/TLSv1.3-Apache

**Comment by janipewter** [comment]
Thanks for the reply, I've already seen those links and I do have it correctly. On further testing I have discovered it is indeed an Apache bug, when configuring a non-default virtualhost to use a different TLS configuration to the default vhost configuration, the TLS 1.3 ciphersuite list is inherited from the default vhost setting, regardless of what you actually configure.

Once I have made a bug report on the Apache bugzilla I will leave the link here.
